libraries/ms-common/src/x86/uart.c

```diff
@@ -1,13 +1,133 @@
 #include "uart.h"
+#include "log.h"
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <arpa/inet.h>
+#include <pthread.h>
+
+typedef struct {
+  int sockfd;
+  int listenfd;
+  int connectfd;
+  struct sockaddr_in addr;
+  UARTStorage storage;
```

Comment:
Why do you store a whole copy in here? If you actually needed the data in `UARTStorage`, it should persist so you could just retain a pointer.

I don't think you actually need all of `UARTStorage`.